go/vt/vttablet/tabletmanager/vdiff/workflow_differ.go

```diff
@@ -119,8 +119,8 @@ func (wd *workflowDiffer) doReconcileExtraRows(dr *DiffReport, maxExtraRowsToCom
 	if dr.ExtraRowsSource == 0 || dr.ExtraRowsTarget == 0 {
 		return nil
 	}
-	matchedSourceDiffs := make([]bool, int(dr.ExtraRowsSource))
-	matchedTargetDiffs := make([]bool, int(dr.ExtraRowsTarget))
+	matchedSourceDiffs := make([]bool, len(dr.ExtraRowsSourceDiffs))
+	matchedTargetDiffs := make([]bool, len(dr.ExtraRowsTargetDiffs))
 	matchedDiffs := int64(0)
 
 	maxRows := int(dr.ExtraRowsSource)
```

Comment:
I'm not sure `maxRows` and the `maxExtraRowsToCompare` function parameter are actually needed here.